docgen/src/guides/sync-url.md

```diff
@@ -0,0 +1,110 @@
+
+# Synchronizing the search with the URL
+
+## Basics
+
+### Initialize the routing
+
+In order to make InstantSearch synchronize the search UI with the
+URL you need to activate a feature called routing. To get you
+started, you can just set the option to `true`:
+
+```javascript
+const search = instantsearch({
+  // don't forget the credentials
+  routing: true,
+});
+```
+
+Setting this option to `true` will indicate to InstantSearch to use
+the default mapping (using the simpleMapping) and output the state
+to the URL (using the history router).
+
+See an [example with the default parameters]().
+
+### Customize the default routing
+
+The example in the previous part is actually equivalent to:
+
+```javascript
+const search = instantsearch({
+  // don't forget the credentials
+  routing: {
+    stateMapping: simpleMapping(),
+    router: historyRouter(),
+  },
+});
+```
+
+The stateMapping lets you customize the UI state before it is serialized and after it has been parsed.
+The default implementation provides no customization options, because each implementation will have
+its own set of customization there.
+
+The mapping is an object that contains two methods:
+ - `stateToRoute(uiState)`
+ -  `routeToState(syncable)`
+
+And those two methods should comply to the following rule: for any `uiState`, `uiState` must be equal in
+value to `routeToState(stateToRoute(uiState))`. This means that any transformation applied by `stateToRoute`
+can be cancelled by `routeToState`.
+
+Let's take an example:
+
+```javascript
+// state mapping that extracts brand from refinementList
+// and put the values in a string
```

Comment:
is this missing the actual code?

Reply:
it's not ready yet :) Don't bother review the content. This is a placeholder